pkg/decoder/delimiter.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package decoder
+
+import (
+	"regexp"
+	"sync"
+	"time"
+)
+
+// Line represents content separated by two '\n'
+type Line struct {
+	content []byte
+}
+
+// NewLine returns a new Line
+func NewLine(content []byte) *Line {
+	return &Line{
+		content: content,
+	}
+}
+
+// Delimiter
+type Delimiter interface {
+	LineChan() chan *Line
+}
+
+// SingleLineDelimiter reads lines from lineChan and uses lineBuffer to send them
+type SingleLineDelimiter struct {
+	lineChan   chan *Line
+	lineBuffer *LineBuffer
+}
+
+// NewSingleLineDelimiter returns a new NewSingleLineDelimiter
+func NewSingleLineDelimiter(outputChan chan *Output) *SingleLineDelimiter {
+	lineChan := make(chan *Line)
+	lineBuffer := NewLineBuffer(outputChan)
+	delimiter := SingleLineDelimiter{
+		lineChan:   lineChan,
+		lineBuffer: lineBuffer,
+	}
+	go delimiter.run()
+	return &delimiter
+}
+
+// LineChan returns lineChan
+func (d *SingleLineDelimiter) LineChan() chan *Line {
+	return d.lineChan
+}
+
+// run reads lines from lineChan to process them
+func (d *SingleLineDelimiter) run() {
+	for line := range d.lineChan {
+		d.process(line)
+	}
+}
+
+// process sends or truncates lines using lineBuffer
+func (d *SingleLineDelimiter) process(line *Line) {
+	lineLen := len(line.content)
+	if lineLen < contentLenLimit {
+		// send line
+		d.lineBuffer.Add(line)
+		d.lineBuffer.Flush()
+	} else {
+		// truncate content
+		d.lineBuffer.AddAndTruncate(line)
```

Comment:
wouldn't we want to flush here too?

Reply:
`AddAndTruncate` already sends a part of the content, maybe the name is not relevant enough, WDYT ?

Reply:
We could rename it to `AddAndTruncateAndFlush` maybe yes

Reply:
or alternatively call `d.lineBuffer.AddAndTruncate(line)` and `d.lineBuffer.Flush()`?